lib/util/template/index.ts

```diff
@@ -201,15 +224,16 @@ export function compile(
   input: CompileInput,
   filterFields = true
 ): string {
-  const data = { ...GlobalConfig.get(), ...input };
+  const data = { ...GlobalConfig.get(), ...input, env: getChildEnv({}) };
   const filteredInput = filterFields ? proxyCompileInput(data) : data;
   logger.trace({ template, filteredInput }, 'Compiling template');
   if (filterFields) {
     const matches = template.matchAll(templateRegex);
     for (const match of matches) {
       const varNames = match[1].split('.');
+      const allowedFieldsList = getAllowedFieldsList();
```

Comment:
move outside the loop

Reply:
otherwise it will be evaluated for each match again, but that set never changes inside the loop